FillableFieldworkAIDriver.lua

```diff
@@ -181,7 +181,10 @@ function FillableFieldworkAIDriver:areFillLevelsOk(fillLevelInfo,isWaitingForRef
 				if fillType == FillType.FERTILIZER or fillType == FillType.LIQUIDFERTILIZER then hasNoFertilizer = true end
 			else
 				if fillType == FillType.SEEDS then hasSeeds = true end
-			end		
+			end
+			if info.fillLevel ~= 0 and fillType == FillType.FERTILIZER or fillType == FillType.LIQUIDFERTILIZER then 
```

Comment:
Why exactly is this needed here?

Reply:
because if there is an empty filUnit that had fertilizer or liquidFertilizer in it the variable `hasNoFertilizer` will be set to true.
And if we find in a later iteration a filUnit with fertilizer or liquid fertilizer it had to be reset to false because now there is fertilizer. 

There is one case that just jumped in my mind where it would not work.
what if the case is inverted? what if it first finds a non-empty fillUnit and then find an empty fillUnit with fertilizer? that still could be a problem but for now I don't have an idea how to solve it.

Reply:
One obvious solution would be to calculate the actual fill levels for fertilizer and seed instead of using booleans, like fertilizerFillLevel and seedFillLevel and check for 0 at the end.

Reply:
thats true.
I will implement that.